src/liquid/components/ld-select/ld-select.tsx

```diff
@@ -687,7 +689,7 @@ export class LdSelect implements InnerFocusable {
         } else {
           let current = document.activeElement
           while (nextLdOption === undefined) {
-            if (current.nextElementSibling?.tagName === 'LD-OPTION-INTERNAL') {
+            if (this.isLdOption(current.nextElementSibling)) {
```

Comment:
Using the new type checking function allows you to remove explicit type casting like here:

```ts
if ((current.nextElementSibling as HTMLElement).hidden) {
```

Found 2 or 3 castings like this in the file that are no longer necessary.